crates/autopilot/src/driver_model.rs

```diff
@@ -56,20 +56,32 @@ pub mod solve {
         primitive_types::{H160, U256},
         serde::{Deserialize, Serialize},
         serde_with::{serde_as, DisplayFromStr},
-        std::collections::BTreeMap,
     };
 
     #[serde_as]
     #[derive(Clone, Debug, Default, Serialize)]
     #[serde(rename_all = "camelCase")]
     pub struct Request {
         pub id: i64,
+        pub tokens: Vec<Token>,
         pub orders: Vec<Order>,
-        #[serde_as(as = "BTreeMap<_, DisplayFromStr>")]
-        pub prices: BTreeMap<H160, U256>,
         pub deadline: DateTime<Utc>,
     }
 
+    #[serde_as]
+    #[derive(Clone, Debug, Serialize)]
+    #[serde(rename_all = "camelCase")]
+    pub struct Token {
+        pub decimals: Option<u8>,
+        pub symbol: Option<String>,
+        pub address: H160,
+        #[serde_as(as = "Option<DisplayFromStr>")]
+        pub price: Option<U256>,
+        #[serde_as(as = "DisplayFromStr")]
+        pub available_balance: U256,
```

Comment:
To me, this is something that is augmented by the `driver` and not essential to the protocol. I would be careful in adding more work to the "auction cutting" for unnecessary things.

Reply:
In other words:

`autopilot::Token`
```
    pub struct Token {
        pub address: H160,
        #[serde_as(as = "Option<DisplayFromStr>")]
        pub price: Option<U256>,
        pub trusted: bool,
    }
```

`driver::Token` after "augmenting":
```
    pub struct Token {
        pub decimals: Option<u8>,
        pub symbol: Option<String>,
        pub available_balance: U256,
        ...autopilot::Token
    }
```

Reply:
Also, I think this can (and likely should) be done in a follow up to not further bloat this PR.

Reply:
Not sure why this should happen in a different PR as this would actually reduce the diff of this PR so we wouldn't have to verify that the changes have been reverted correctly.